src/core/init.js

```diff
@@ -50,19 +70,22 @@ var rootjQuery,
 					// Intentionally let the error be thrown if parseHTML is not present
 					jQuery.merge( this, jQuery.parseHTML(
 						match[ 1 ],
-						context && context.nodeType ? context.ownerDocument || context : document,
+						context && context.nodeType ?
+							context.ownerDocument || context :
+							document,
 						true
 					) );
 
 					// HANDLE: $(html, props)
-					if ( rsingleTag.test( match[ 1 ] ) && jQuery.isPlainObject( context ) ) {
+					if ( rsingleTag.test( htmlDeserialized || match[ 1 ] ) &&
```

Comment:
Just a small size optimization: 

TT stringify (we rely on that when creating `match` from `selector` on line 47). 

We could leverage that and call `rSingleTag.test(match[1])` directly, that way `htmlDeserialized` variable is not needed at all - unless there is some linter check that prevents calling RegExp.test with a non-string.

That said, this change would enable an interesting trick, in which you pass to jQuery an object that stringifies differently on every attempt - but I'm not sure jQuery aims to be resilient to this kind of attacks from a code that calls jQuery (there's probably 100 more vectors like that).

Reply:
Good point, I'll try that. I didn't consider that `RegExp#test` stringifies its input.